gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java

```diff
@@ -193,6 +212,52 @@ public void onFocusChange(Field field, boolean hasFocus) {
     }
   }
 
+  void initPhotoSelector(String fieldId) {
+    /*
+     * Didn't subscribe this with Fragment's lifecycle because we need to retain the disposable
+     * after the fragment is destroyed (for activity result)
+     */
+    disposeOnClear(
```

Comment:
disposeOnClear will remove the subscription as soon as the ViewModel is removed, and we can't assume the ViewModel will stick around after the Fragment is destroyed. Do we need this because the photo picker Intent causes the fragment to go away? If so, can we launch image picker synchronously, and only subscribe to image picker results?

Reply:
Initially, I tried subscribing the RxChain in Fragment using it's lifecycle. But during launch of `Intent`, the fragment gets destroyed and so does the disposable.

> can we launch image picker synchronously, and only subscribe to image picker results?

If we separate this logic, then how would we pass the requesting `fieldId` to update the response later?
Since one observation can have multiple photo fields, we have to know the requesting fieldId in advance. Thoughts?

Reply:
Can you use the Bundle stored in the Intent's "extras" to pass the field id through to the photo picker and back again?

Reply:
This method can be called multiple times, accumulating subscriptions until the view model is destroyed. If we can't decouple request from the activity response, please add a TODO here to trigger this from a switchMap or to find another way to split the stream in two to avoid accumulating subscriptions.